devito/yask/function.py

```diff
@@ -148,3 +167,35 @@ def _arg_defaults(self, alias=None):
 class TimeFunction(function.TimeFunction, Function):
 
     from_YASK = True
+
+    @classmethod
+    def __indices_setup__(cls, **kwargs):
+        indices = list(function.TimeFunction.__indices_setup__(**kwargs))
+        # Never use a SteppingDimension in the yask backend: it is simply
+        # unnecessary and would only complicate things when creating dummy
+        # grids
+        indices[cls._time_position] = kwargs['grid'].time_dim
+        return tuple(indices)
+
+    def _arg_defaults(self, alias=None):
+        args = super(TimeFunction, self)._arg_defaults(alias=alias)
+        # This is a little hack: a TimeFunction originally meant to be accessed
+        # via modulo buffered iteration should never impose anything on the time
+        # dimension
+        if type(self.save) is not int:
```

Comment:
I'm not sure that this is a very elegant condition...  e.g. if someone accidentally computes save=(8/2), this will fail and confuse them.

Should we just use the truthiness of `save`, instead?  (and also complain bitterly if it's truthy and not an integer...  but that's presumably elsewhere).

Reply:
Just to be clear. ``self.save`` is a type. In particular, its value can be either `int` or `NoneType`; see here:
https://github.com/opesci/devito/blob/master/devito/function.py#L660

Hence, you basically just spotted a bug here: I shouldn't be using ``type(self.save)``, but rather ``self.save``. I'm fixing this !

Reply:
Eek.  Ok!